libraries/c_sdk/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -838,6 +854,45 @@ void _IotMqtt_DecrementConnectionReferences( _mqttConnection_t * pMqttConnection
 
 /*-----------------------------------------------------------*/
 
+static uint32_t getTimeMs( void )
+{
+    uint32_t timeMs;
+
+    return ( uint32_t ) IotClock_GetTimeMs();
```

Comment:
IotClock_GetTimeMs returns a 64 bit value, directly type casting it to 32 bit will directly take only the lower order bits. I think it may still be ok as MQTT library is handling overflows. I think even then having the elapsed time calculated in the shim would be a better choice. Please refer to MQTT demos in development branch in C-SDK